test/e2e/specs/editor/various/post-title.spec.js

```diff
@@ -0,0 +1,179 @@
+				content: 'Hello world',
+				status: 'publish',
+			} );
+
+			await admin.visitAdminPage(
+				'post.php',
+				`post=${ postId }&action=edit`
+			);
+
+			await page.evaluate( () => {
+				window.wp.data
+					.dispatch( 'core/preferences' )
+					.set( 'core/edit-post', 'welcomeGuide', false );
+
+				window.wp.data
+					.dispatch( 'core/preferences' )
+					.set( 'core/edit-post', 'fullscreenMode', false );
+			}, false );
+
+			// switch Editor to code view mode
+			// Open code editor
+			await pageUtils.pressKeys( 'secondary+M' ); // Emulates CTRL+Shift+Alt + M => toggle code editor
+
+			// Check we're in Code view mode.
+			await expect(
+				editor.canvas.getByRole( 'heading', {
+					name: 'Editing code',
+				} )
+			).toBeVisible();
+
+			const codeViewPageTitleField = editor.canvas.getByRole( 'textbox', {
+				name: 'Add title',
+			} );
+
+			// Check that the pageTitleField has the raw HTML
+			await expect( codeViewPageTitleField ).toHaveText(
+				'I am <em>emphasis</em> I am <strong>bold</strong> I am <a href="#">anchor</a>'
+			);
+		} );
+
+		test( 'should strip HTML tags when pasting string of HTML into the post title field in Visual mode', async ( {
+			editor,
+			admin,
+			pageUtils,
+		} ) => {
+			await admin.createNewPost();
+
+			const pageTitleField = editor.canvas.getByRole( 'textbox', {
+				name: 'Add title',
+			} );
+
+			await expect( pageTitleField ).toBeFocused();
+
+			pageUtils.setClipboardData( {
+				html: 'I am <em>emphasis</em> I am <strong>bold</strong> I am <a href="#">anchor</a>',
+			} );
+			await pageUtils.pressKeys( 'primary+v' );
+
+			await expect( pageTitleField ).toHaveText(
+				'I am emphasis I am bold I am anchor'
+			);
+
+			// Check the HTML elements have been stripped and are not rendered.
+			await expect( pageTitleField.locator( 'css=em' ) ).toBeHidden();
+
+			await expect( pageTitleField.locator( 'css=strong' ) ).toBeHidden();
+
+			await expect( pageTitleField.locator( 'css=a' ) ).toBeHidden();
+		} );
+
+		test( 'should retain HTML tags when pasting string of HTML into the post title field in Code view mode', async ( {
+			editor,
+			admin,
+			pageUtils,
+		} ) => {
+			await admin.createNewPost();
+
+			// switch Editor to code view mode
+			// Open code editor
+			await pageUtils.pressKeys( 'secondary+M' ); // Emulates CTRL+Shift+Alt + M => toggle code editor
+
+			// Check we're in Code view mode.
+			await expect(
+				editor.canvas.getByRole( 'heading', {
+					name: 'Editing code',
+				} )
+			).toBeVisible();
+
+			const pageTitleField = editor.canvas.getByRole( 'textbox', {
+				name: 'Add title',
+			} );
+
+			pageUtils.setClipboardData( {
+				plainText:
+					'I am <em>emphasis</em> I am <strong>bold</strong> I am <a href="#">anchor</a>',
+				html: 'I am <em>emphasis</em> I am <strong>bold</strong> I am <a href="#">anchor</a>',
+			} );
+
+			// focus on the title field
+			await pageTitleField.focus();
+
+			await pageUtils.pressKeys( 'primary+v' );
```

Comment:
🆘 For some reason this does not work. Pasting in code view mode just does not happen. I'm missing something but I'm not sure what it is...

Reply:
@jeryj I know you are really good at debugging these tests. Any ideas would be hugely appreciated 🙏

Reply:
I looked into this for awhile and couldn't isolate it. I feel like there's an issue between `emulateClipboard` and the code editor view. I can `page.keyboard.type` in the code editor, but the paste doesn't work. The paste command is being fired correctly and the clipboard data text is being set correctly. 

@kevin940726 and @Mamaduka - do y'all have any ideas on this one or tips for debugging it further?

Reply:
Dispatching `paste` events only works for elements that handle the `paste` event internally. So regular `<textarea>` that doesn't bind any `paste` listeners won't work. I only just knew that too, so it's also a surprise to me 🤦‍♂️.

I tried using the Clipboard API and playwright's `Meta+KeyV` but it only works in headed mode in some cases. I don't know if it's possible to feature detect it before the emulation either. There's a long-standing [issue](https://github.com/microsoft/playwright/issues/8114) upstream too.

I don't think there's a perfect catch-all solution, but I might be able to come up with one that's "good enough" in most cases. Stay tuned! 🤞

Reply:
Opened a PR in https://github.com/WordPress/gutenberg/pull/55030 🤞 .

Reply:
I've `skip` my test and will reinstate once your PR is merged 🤞